sssd_test_framework/utils/local_users.py

```diff
@@ -421,3 +440,176 @@ def remove_members(self, members: list[LocalUser]) -> LocalGroup:
+        if isinstance(item, list):
+            result = ", ".join(str(x) for x in item)
+        else:
+            result = str(item)
+        return result
+
+    def add(
+        self,
+        *,
+        user: str | LocalUser | LocalGroup | list[str | LocalUser | LocalGroup],
+        host: str | list[str] = "ALL",
+        command: str | list[str] = "ALL",
+        option: str | list[str] | None = None,
+        runasuser: str | LocalUser | list[str | LocalUser] | None = None,
+        runasgroup: str | LocalGroup | list[str | LocalGroup] | None = None,
+        order: int | None = None,
+        nopasswd: bool | None = None,
+    ) -> LocalSudoRule:
+        """
+        Create new sudo rule.
+
+        :param user: sudoUser attribute, defaults to None
+        :type user: str | LocalUser | LocalGroup | list[str | LocalUser | LocalGroup]
+        :param host: sudoHost attribute, defaults to None
+        :type host: str | list[str], defaults to ALL
+        :param command: sudoCommand attribute
+        :type command: str | list[str], defaults to ALL
+        :param option: sudoOption attribute, defaults to None
+        :type option: str | list[str] | None, optional
+        :param runasuser: sudoRunAsUser attribute, defaults to None
+        :type runasuser: str | LocalUser | list[str | LocalUser] | None, optional
+        :param runasgroup: sudoRunAsGroup attribute, defaults to None
+        :type runasgroup: str | LocalGroup | list[str | LocalGroup] | None, optional
+        :param order: sudoOrder attribute, defaults to None
+        :type order: int | None, optional
+        :param nopasswd: If true, no authentication is required (NOPASSWD), defaults to None (no change)
+        :type nopasswd: bool | None, optional
+        :return: New sudo rule object.
+        :rtype: LocalSudoRule
+        """
+        orderstr = f"{order:02d}" if order is not None else str(len(self.util._sudorules))
+        if self.filename is None:
+            self.filename = f"{orderstr}_{self.name}"
+
+        # Remember arguments so we can use them in modify if needed
+        self.__rule = dict[str, Any](
+            user=user,
+            host=host,
+            command=command,
+            option=option,
+            runasuser=runasuser,
+            runasgroup=runasgroup,
+            order=order,
+            nopasswd=nopasswd,
+        )
+        run_as_str = ""
+        if runasuser or runasgroup:
+            run_as_str += "("
+            if runasuser:
+                run_as_str += LocalSudoRule._format_list(runasuser)
+            if runasgroup:
+                run_as_str += f":{LocalSudoRule._format_list(runasgroup)}"
+            run_as_str += ")"
+        user_str = LocalSudoRule._format_list(user)
+        if isinstance(host, list):
+            host_str = ", ".join(host)
+        else:
+            host_str = host
+        host_str += "="
+        tagspec_str = "NOPASSWD:" if nopasswd else ""
+        if isinstance(command, list):
```

Comment:
You can use `LocalSudoRule._format_list(command)` here.

If the reason for not using it is that "ALL" must be the default value when command is `None`, that can never happen because the declaration of `command` includes the default value `"ALL"`.

Well, this could happen if the caller explicitly provides the value `None`, which is an error.

Reply:
Done.